Document OVNKubernetes (#422)
submariner-io/submariner-website: src/content/getting-started/architecture/route-agent/_index.en.md

running on the other worker nodes in the local cluster connect by establishing a VXLAN tunnel with the VTEP of the active Gateway Engine
node. The MTU of the VXLAN tunnel is configured based on the MTU of the default interface on the host minus the VXLAN overhead.
For **_kubeproxy-iptables_** based implementations it creates VxLAN tunnels from
the node to the cluster's active Gateway node, as well as routing
to allow normal Pod and host-network connectivity to remote clusters.
When a Gateway fails and HA failover happens the route-agent handler will update
the VxLAN tunnels as well as the routes.

Route Agents use `Endpoint` resources synced from other clusters to configure routes and to program the necessary iptables rules to enable
full cross-cluster connectivity.
For **_OVNKubernetes_** host-network routing is configured on all nodes, and
on the active Gateway node forwarding is configured between the **_ovn-k8s-gw0_**
and the cable interface.

When the active Gateway Engine fails and a new Gateway Engine takes over, Route Agents will automatically update the route tables on each

> node to point to the new active Gateway Engine node.

Reviewer: A lot of this detail has been removed.  I assume it still applies (for kubeproxy-iptables anyway) in which case I think we should still retain it and work in the OVNKubernetes details around it.

Author: Ack, I have restored that part.

> _submariner-operator_ when creating the networkplugin-syncer deployment.

Author: I believe it's important for debugging, somebody trying to look at the architecture is probably trying to debug an issue. The route agent will behave wildly different based on the network plugin. I think we should at least give a hint about that.

> _submariner-operator_ when creating the networkplugin-syncer deployment.

Reviewer: I disagree - this is a high-level architecture doc - low-level debugging info belongs in a troubleshooting doc.

> _submariner-operator_ when creating the networkplugin-syncer deployment.

Author: I have removed the reference to the precise env var, but retained the phrase.

---
title: "Route Agent"
---

The Route Agent component runs on every node in each participating cluster.
It is responsible for setting up the necessary host network elements on top

> of the existing network plugin.

Reviewer: ```suggestion
of the existing Kubernetes CNI plugin.
```

> The Route Agent receives the detected K8s network plugin as part of its

Reviewer: ```suggestion
The Route Agent receives the detected Kubernetes network plugin as part of its
```

Author: Done.

> The Route Agent receives the detected Kubernetes CNI plugin as part of its

Reviewer: Since we mention the full _Kubernetes CNI plugin_ above...

```suggestion
The Route Agent receives the detected CNI plugin as part of its
```

configuration.

> ## kubeproxy-iptables

Reviewer: The K8s docs seem to consistently refer to it as kube-proxy.

```suggestion
## kube-proxy
```

Author: Yes, but there is also kubeproxy IPVS that we don't support, I believe we should keep the distinction I guess kube-proxy iptables

Author: done +/- ;)

> For `kubeproxy-iptables` based implementations it is responsible for setting

Reviewer: ```suggestion
For CNI plugins that utilize kube-proxy IP tables, the Route Agent is responsible for setting
```

Author: Done.

> For CNI plugins that utilize kube-proxy IP tables, the Route Agent is responsible for

Reviewer 2: ```suggestion
For CNI plugins that utilize kube-proxy in iptables mode, the Route Agent is responsible for
```

Author: Done.

up VXLAN tunnels and routing the cross cluster traffic from the node to the
cluster’s active Gateway Engine which subsequently sends the traffic to the
destination cluster.

When running on the same node as the active Gateway Engine, Route Agent
creates a VXLAN VTEP interface to which Route Agent instances running on
the other worker nodes in the local cluster connect by establishing a VXLAN
tunnel with the VTEP of the active Gateway Engine node. The MTU of the VXLAN
tunnel is configured based on the MTU of the default interface on the host
minus the VXLAN overhead.

Route Agents use `Endpoint` resources synced from other clusters to configure
routes and to program the necessary iptables rules to enable full cross-cluster
connectivity.

When the active Gateway Engine fails and a new Gateway Engine takes over,
Route Agents will automatically update the route tables on each node to point to
the new active Gateway Engine node.

## OVN Kubernetes

> For `OVN Kubernetes` host-network routing is configured on all nodes, and
> on the active Gateway node IP forwarding is configured between the `ovn-k8s-gw0`
> and the cable interface.

Reviewer: ```suggestion
For the OVN Kubernetes CNI plugin,  host network routing is configured on all nodes and,
on the active Gateway node, IP forwarding is configured between the `ovn-k8s-gw0`
and cable interface.
```

Author: Done.

> and cable interface.

Reviewer: ```suggestion
and cable interfaces.
```
